src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -468,7 +473,7 @@ public static string Join(string? path1, string? path2, string? path3, string? p
             return Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan(), path4.AsSpan());
         }
 
-        public static string Join(params string[] paths)
+        public static string Join(params string?[] paths)
```

Comment:
Maybe we want to flow this annotation as part of a separate PR that flows into corefx, so that we can annotate it in the ref assembly on the first push? Or it doesn't make sense because we will still have to re-iterate on the ref assembly to annotate the attributes?

Reply:
> Or it doesn't make sense because we will still have to re-iterate on the ref assembly to add attributes?

I was thinking that, but I can also fix it in a separate PR, that's no problem.

Reply:
Fixed in a separate PR.